graphsdk/src/androidTest/java/com/microsoft/graph/functional/ODataTests.java

```diff
@@ -47,12 +38,64 @@ public void testOpenExtensions() {
+        Extension newExtension = testBase.graphClient.getMe().getExtensions().buildRequest().post(extension);
+        assertEquals(extension.getAdditionalDataManager().get("theme"), newExtension.getAdditionalDataManager().get("theme"));
 
-//        Extension newExtension = testBase.graphClient.getMe().getExtensions().buildRequest().post(extension);
-//
-//        assertEquals(extension.getAdditionalDataManager().get("theme"), newExtension.getAdditionalDataManager().get("theme"));
-//
-//        testBase.graphClient.getMe().getExtensions(newExtension.id).buildRequest().delete();
+        testBase.graphClient.getMe().getExtensions(newExtension.id).buildRequest().delete();
+    }
+
+    @Test
+    public void testSchemaExtensions() {
+        SchemaExtension extension = new SchemaExtension();
+        extension.id = "schematest";
+        extension.description = "Android Graph SDK test";
+        List<String> targets = new ArrayList<>();
+        targets.add("Group");
+        extension.targetTypes = targets;
+
+        ExtensionSchemaProperty prop = new ExtensionSchemaProperty();
+        prop.name = "courseId";
+        prop.type = "Integer";
+
+        ExtensionSchemaProperty prop2 = new ExtensionSchemaProperty();
+        prop2.name = "courseName";
+        prop2.type = "String";
+
+        List<ExtensionSchemaProperty> properties = new ArrayList<>();
+        properties.add(prop);
+        properties.add(prop2);
+        extension.properties = properties;
+
+        SchemaExtension newExtension = testBase.graphClient.getSchemaExtensions().buildRequest().post(extension);
+        assertEquals(extension.description, newExtension.description);
+
+        SchemaExtension patchExtension = new SchemaExtension();
+        List<ExtensionSchemaProperty> patchProperties = new ArrayList<>();
+        ExtensionSchemaProperty patchProperty = new ExtensionSchemaProperty();
+        patchProperty.name = "newItem";
+        patchProperty.type = "String";
+        patchProperties.add(prop);
+        patchProperties.add(prop2);
+        patchProperties.add(patchProperty);
+
+        patchExtension.properties = patchProperties;
+
+        testBase.graphClient.getSchemaExtensions(newExtension.id).buildRequest().patch(patchExtension);
+        SchemaExtension updatedExtension = testBase.graphClient.getSchemaExtensions(newExtension.id).buildRequest().get();
+
+        boolean foundUpdatedProperty = false;
+        for (ExtensionSchemaProperty updatedProperty : updatedExtension.properties) {
+            if (updatedProperty.name.equals(patchProperty.name)) {
+                assertEquals(patchProperty.type, updatedProperty.type);
+                foundUpdatedProperty = true;
+                break;
+            }
+        }
+        if (!foundUpdatedProperty) {
+            Assert.fail("Patch failed on Schema Extension");
+        }
+
+        testBase.graphClient.getSchemaExtensions(newExtension.id).buildRequest().delete();
```

Comment:
If the assert fails, it looks like we don't attempt to delete the extension. Maybe move this to a catch block?